applications/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -193,16 +193,18 @@ static int ble_adv_start_directed(const bt_addr_le_t *addr, bool fast_adv)
 
 	if (fast_adv) {
 		LOG_INF("Use fast advertising");
-		adv_param = *BT_LE_ADV_CONN_DIR;
+		adv_param = *BT_LE_ADV_CONN_DIR(addr);
 	} else {
-		adv_param = *BT_LE_ADV_CONN_DIR_LOW_DUTY;
+		adv_param = *BT_LE_ADV_CONN_DIR_LOW_DUTY(addr);
 	}
 
 	adv_param.id = cur_identity;
 
-	struct bt_conn *conn = bt_conn_create_slave_le(addr, &adv_param);
+	struct bt_conn *conn = NULL;
 
-	if (conn == NULL) {
+	int rc = bt_le_adv_start(&adv_param, NULL, 0, NULL, 0);
```

Comment:
`err` instead of `rc` - let's stay consistent.
Also `conn` is not used so can be removed.

Reply:
it's even worse - it will always lead to `-ENOMEM`!

Reply:
Correct, also don't return -ENOMEM anymore, forward the error code.

```
int err = bt_le_adv_start(&adv_param, NULL, 0, NULL, 0);
if (err ) {
       return err;
}
```

Reply:
ok, done